[CMake] Add WebKit/Platform/Cairo.cmake

fujii opened the pull request:
#### 62e7a0a1756386de764e1ca20ccfd96ca88c3f27
<pre>
[CMake] Add WebKit/Platform/Cairo.cmake
<a href="https://bugs.webkit.org/show_bug.cgi?id=284142">https://bugs.webkit.org/show_bug.cgi?id=284142</a>

Reviewed by Don Olmstead.

Consolidated Cairo CMake code into Cairo.cmake in WebKit module.

* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/PlatformWin.cmake:
* Source/WebKit/Platform/Cairo.cmake: Added.

Canonical link: <a href="https://commits.webkit.org/287461@main">https://commits.webkit.org/287461@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3add28e0369a9bc1ab3fef3966f3b75d709c1fb0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/79818 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/58819 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/33217 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/84343 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/30820 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/67893 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/7109 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/62388 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/20228 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/82886 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/52449 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/72691 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/42695 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/49789 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/29268 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/70916 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/27313 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/85769 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/7049 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/4941 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/70652 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/7223 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/68531 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/69891 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/13896 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/12815 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12331 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/7008 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/12586 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/6868 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/10379 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/8675 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

donny-dont commented on Source/WebKit/Platform/Cairo.cmake:
```diff
@@ -0,0 +1,27 @@
+    list(APPEND WebKit_SOURCES
+        UIProcess/cairo/BackingStoreCairo.cpp
+    )
+endif ()
```
I think you should just keep this in `WC.cmake` since its required there.

fujii replied:
Fixed.

donny-dont commented on Source/WebKit/PlatformPlayStation.cmake:
```diff
@@ -130,30 +130,11 @@ if (ENABLE_WEBDRIVER AND USE_WPE_BACKEND_PLAYSTATION)
-
-        UIProcess/Automation/cairo/WebAutomationSessionCairo.cpp
-
-        UIProcess/cairo/BackingStoreCairo.cpp
```
Just leave this part in here.

fujii replied:
Fixed.

donny-dont reviewed (approved):
LGTM. Don't have a strong opinion on my comments so make your decision there.

fujii committed:
[CMake] Add WebKit/Platform/Cairo.cmake
https://bugs.webkit.org/show_bug.cgi?id=284142

Reviewed by Don Olmstead.

Consolidated Cairo CMake code into Cairo.cmake in WebKit module.

* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/PlatformWin.cmake:
* Source/WebKit/Platform/Cairo.cmake: Added.

Canonical link: https://commits.webkit.org/287461@main
